src/_common/fireside/post/post-model.ts

```diff
@@ -30,6 +31,47 @@ interface FiresidePostPublishedPlatform {
 	url: string;
 }
 
+export function isInCommunityPinContext(post: FiresidePost, route: Route) {
+	if (post.communities.length > 0) {
+		const communityLink = post.communities[0];
+		const community = communityLink.community;
+		const channelTitle = communityLink.channel!.title;
+		if (
+			route.name === 'communities.view.overview' &&
+			route.params.path === community.path &&
+			route.params.channel === channelTitle
+		) {
+			return true;
+		}
+	}
+	return false;
+}
+
+export function isInPinContext(post: FiresidePost, route: Route) {
+	// For the pin option to show, the user has to look at the correct feed for the post's context.
+	// A game post can only be pinned from the game's overview or game's dash feeds.
+	// A user post can only be pinned to the user from the user's overview
+	// 		OR a user post can be pinned to a community channel from a community channel, NOT featured though.
+
+	if (post.game instanceof Game) {
+		return (
+			(route.name === 'discover.games.view.overview' ||
+				route.name === 'dash.games.manage.devlog') &&
+			route.params.id.toString() === post.game.id.toString()
+		);
+	}
+
+	if (isInCommunityPinContext(post, route)) {
+		return true;
+	}
+
+	if (route.name === 'profile.overview' && route.params.username === post.user.username) {
```

Comment:
should also check youre looking at the active tab. i think it'll be weird to offer to pin a draft or scheduled post, even if we make that not show up.

Reply:
It checks this in the extra menu.